Smaller version: 6 lines removed

MoiTux opened the pull request:
When reading the article I directly see that the `for` in `Match` could be optimized.
Then I see that the same things could be done with `matchStar`.

I think the first change (in `Match`) is OK and its still readable.
For the second one (in `matchStar`), it might remove some readability ?

Commit:
Smaller version: 4 lines removed

When reading the article I directly see that the `for` in `Match` could be optimized.
Then I see that the same things could be done with `matchStar`.

I think the first change (in `Match`) is OK and its still readable.
For the second one (in `matchStar`), it might remove some readability ?

benhoyt commented:
Hi @MoiTux, thanks for this -- that's clever! I've added an [update](https://github.com/benhoyt/benhoyt.github.com/commit/074b999136be9a7951b772e3d89aac61621f1500) to the [article](https://benhoyt.com/writings/rob-pike-regex/) mentioning your improvement. I won't merge this in (so people don't get the idea this is my idea :-) and keep this PR open to link to it there.

Commit:
Just for the fun of it, let's remove an other 2 lines.

This can be farfetch and even if it's valid go code, it's not a common one.
I will not use this in a real situation.

MoiTux commented:
I wasn't sure what was the best way to submit a solution, so this is fine with me.

Thanks for the added part in the article.  

BTY I just push an other version on my fork with 2 more line removed: this one is debatable I think :D 
https://github.com/benhoyt/repike/commit/c89b143019894020344b55ad538d19f213bb8b7c

benhoyt commented:
Nice! Updated the article text to mention that part.
